docs/sphinx/examples/cpp/providers/nvqc_sample.cpp

```diff
@@ -1,23 +1,23 @@
 // Compile and run with:
 // ```
-// nvq++ --target nvcf nvcf_sample.cpp -o out.x
+// nvq++ --target nvqc nvqc_sample.cpp -o out.x
 // ./out.x
 // ```
-// Assumes a valid NVCF API key and function ID have been stored in environment
-// variables or `~/.nvcf_config` file. Alternatively, they can be set in the
+// Assumes a valid NVQC API key and function ID have been stored in environment
```

Comment:
Why does the function ID need to be stored as environment variable?
The CUDA Quantum runtime should take care of that automatically. Since this is a public example, I would remove any mention of the function-id configuration option - at this point, we do not expect that anyone would use a custom function, do we?

Reply:
Fixed in https://github.com/NVIDIA/cuda-quantum/pull/1247/commits/ae21a9b4165762420178cb7cbfa3a4cdf89892af:

- Query NVCF to get available functions. 

- Client to track the number of GPUs of each available deployed function. Add `ngpus` option.

- Update examples to get rid of function Id.

Side note: I figured that it's more convenient to implement these in this PR rather than a separate PR.